packages/dx-react-grid/src/plugins/table-selection.jsx

```diff
@@ -71,7 +71,7 @@ export class TableSelection extends React.PureComponent {
                   <Cell
                     {...params}
                     row={params.tableRow.row}
-                    selected={selection.indexOf(params.tableRow.rowId) !== -1}
+                    selected={new Set(selection).has(params.tableRow.rowId)}
```

Comment:
Hm, I don't think that it is needed here. It will check the row id only once. We may spend more memory and CPU to construct a set that will be removed soon.